medallion/backends/memory_backend.py

```diff
@@ -122,14 +123,37 @@ def _pop_expired_sessions(self):
 
         for item in expired_ids:
             self.next.pop(item)
-
+    def collections_manifest_check(self):
+        """
+        Checks collections for proper manifest, if objects are present in a collection, a manifest should be present with 
+        an entry for each entry in objects
+        """
+        for key, api_root in self.data:
+            for collection in api_root.get('collections', [])
+                if not collection.get('objects'):
+                    continue
+                if 'manifest' not in collection:
+                    raise InitializationError("Collection {} manifest is missing".format(collection['id']), 408)
+                else if not collection['manifest']:
+                    raise InitializationError("Collection {} with objects has an empty manifest".format(collection['id']), 408)
+                for obj in collection.get('objects', []):
+                    obj_time=find_att(obj)
+                    obj_man_paired = False
+                    for man in collection['manifest']:
+                        man_time = find_att(man)
+                        if obj['id'] == man['id'] and obj_time == man_time:
+                            obj_man_paired = True
+                    if not obj_man_paired: 
+                        raise InitializationError("Object with id {} last modified on {} is missing a manifest".format(obj['id'], obj_time), 408)
```

Comment:
It won't always be the last modified timestamp, depending on the type of object - see the source code for `find_att()`.
```suggestion
                        raise InitializationError("Object with id {} from {} is missing a manifest".format(obj['id'], obj_time), 408)
```